api/Controllers/CategoryController.cs

```diff
@@ -7,70 +7,76 @@
-        {
-            var category = await _categoryService.GetCategory(categoryId);
-            if(category == null) {
-                return NotFound();
-            }
-            var categoryDto = _categoryService.TransformCategoryDtoForOutput(_mapper.Map<CategoryDto>(category));
-            return Ok(categoryDto);
-        }
-        
-        [HttpGet]
-        public async Task<IActionResult> GetCategoryList() 
-        {
-            var categoryList = await _categoryService.GetCategoryList();
-            if(categoryList == null) {
-                return NotFound();
-            }
-            var categoryDtoList = _mapper.Map<IList<CategoryDto>>(categoryList);
-            return Ok(_categoryService.TransformCategoryDtoListForOutput(categoryDtoList));
-        }
+    [HttpPost]
+    public async Task<IActionResult> PostCategory(CategoryDto category)
+    {
+      Category newCategory = await _categoryService.CreateCategory(category);
 
-        [HttpPost]
-        public async Task<IActionResult> PostCategory(CategoryDto category) 
-        {   
-            Category newCategory = await _categoryService.CreateCategory(category);
-            
-            var createdCategory = await _categoryService.PostCategory(newCategory);
-            if(createdCategory == null) {
-                return BadRequest();
-            }
-            return Created(nameof(GetCategory), _mapper.Map<CategoryDto>(createdCategory));
-        }
+      var createdCategory = await _categoryService.PostCategory(newCategory);
+      if (createdCategory == null)
+      {
+        return BadRequest();
+      }
+      return Created(nameof(GetCategory), _mapper.Map<CategoryDto>(createdCategory));
+    }
 
-        [HttpDelete]
-        public async Task<ActionResult> DeleteCategory(int categoryId) {
-            var removedCategory = await _categoryService.DeleteCategory(categoryId);
-            return removedCategory ? Ok() : BadRequest();
-        }
+    [HttpDelete]
+    public async Task<ActionResult> DeleteCategory(int categoryId)
+    {
+      var removedCategory = await _categoryService.DeleteCategory(categoryId);
+      return removedCategory ? Ok() : BadRequest();
+    }
 
-        [HttpPut]
-        public async Task<ActionResult> PutCategory(int categoryId, CategoryDto updatedCategory) {
-            Category categoryToPut = await _categoryService.CreateCategory(updatedCategory);
+    [HttpPut]
+    public async Task<ActionResult> PutCategory(CategoryDto updatedCategory)
+    {
+      Category categoryToPut = await _categoryService.CreateCategory(updatedCategory);
 
-            var newCategory = await _categoryService.PutCategory(categoryId, categoryToPut);
-            if(newCategory == null) {
-                return BadRequest();
-            }
-            return Created(nameof(GetCategory), _mapper.Map<CategoryDto>(newCategory));
-        }
+      var newCategory = await _categoryService.PutCategory(updatedCategory.Id, categoryToPut);
```

Comment:
I have to admit that this works, but can't he take the id directly from _categoryToPut_? And then you have to send only one parameter like this __categoryService.PutCategory(categoryToPut);_
And in _PutCategory_ from _categoryService_ you can change to take the id from _categoryToPut_